spec.html

```diff
@@ -13019,18 +13043,25 @@ <h1>Runtime Semantics: PropertyDefinitionEvaluation</h1>
         <emu-alg>
           1. Let _propKey_ be the result of evaluating |PropertyName|.
           1. ReturnIfAbrupt(_propKey_).
-          1. If IsAnonymousFunctionDefinition(|AssignmentExpression|) is *true*, then
+          1. If this |PropertyDefinition| is contained within a |Script| that is being evaluated for JSON.parse (see step <emu-xref href="#step-json-parse-parse"></emu-xref> of JSON.parse), then
+            1. Let _isProtoSetter_ be *false*.
+          1. Else, If _propKey_ is the String value *"__proto__"* and if IsComputedPropertyKey(|PropertyName|) is *false*, then
+            1. Let _isProtoSetter_ be *true*.
+          1. Else,
+            1. Let _isProtoSetter_ be *false*.
+          1. If IsAnonymousFunctionDefinition(|AssignmentExpression|) is *true* and _isProtoSetter_ is *false*, then
             1. Let _propValue_ be NamedEvaluation of |AssignmentExpression| with argument _propKey_.
           1. Else,
             1. Let _exprValueRef_ be the result of evaluating |AssignmentExpression|.
             1. Let _propValue_ be ? GetValue(_exprValueRef_).
+          1. If _isProtoSetter_ is *true*, then
+            1. If Type(_propValue_) is either Object or Null, then
+              1. Return ! _object_.[[SetPrototypeOf]](_propValue_).
+            1. Return NormalCompletion(~empty~).
```

Comment:
I really would prefer if this was a `ThrowCompletion`, but that would be a breaking change, which is why I suggested adding `#proto` in <https://github.com/tc39/ecma262/issues/2088>.